lib/discover/manager.go

```diff
@@ -34,8 +34,11 @@ import (
 type Manager interface {
 	FinderService
 	ChildErrors() map[string]error
+	DiscoveryStatus() map[string]DiscoveryStatusEntry
```

Comment:
As `ChildErrors` returns all children/finders/discoverers, also if there's no error, I'd just remove it and only have a single method that returns status.

Naming wise we usually leave out `Discovery` here, as it's redundant with the package name - same as it isn't `DiscoveryManager`. Just `Status` doesn't convey that it's multiple ones. As we do already use the term `Finder` in exported parts of the package: maybe `FinderStatus`.